include/matx/transforms/solver_common.h

```diff
@@ -0,0 +1,297 @@
+//
+// 1. Redistributions of source code must retain the above copyright notice, this
+//    list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright notice,
+//    this list of conditions and the following disclaimer in the documentation
+//    and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the copyright holder nor the names of its
+//    contributors may be used to endorse or promote products derived from
+//    this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+/////////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+
+#ifdef MATX_EN_NVPL
+  #include <nvpl_lapack.h>
+  using lapack_int_t = nvpl_int_t;
+  using lapack_scomplex_t = nvpl_scomplex_t;
+  using lapack_dcomplex_t = nvpl_dcomplex_t;
+#else
+  using lapack_int_t = int64_t;
+#endif
+
+namespace matx {
+
+/* Parameter enums */
+
+// Which part (lower or upper) of the dense matrix was filled
+// and should be used by the function
+enum class SolverFillMode {
+  UPPER,
+  LOWER
+};
+
+enum class EigenMode {
+  NO_VECTOR, // Only eigenvalues are computed
+  VECTOR     // Both eigenvalues and eigenvectors are computed
+};
+
+// SVD job options for computing columns of U (jobu) and rows of VT (jobvt)
+enum class SVDJob {
+  ALL,     // For jobu: All M columns of U are computed
+           // For jobvt: All N rows of V^T are computed
+  REDUCED, // For jobu: The first min(m,n) columns of U are computed
+           // For jobvt: The first min(m,n) rows of V^T are computed
+  NONE     // For jobu: No columns of U are computed
+           // For jobvt: No rows of V^T are computed
+};
+
+namespace detail {
+
+__MATX_INLINE__ char SVDJobToChar(SVDJob job) {
+  switch (job) {
+    case SVDJob::ALL:
+      return 'A';
+    case SVDJob::REDUCED:
+      return 'S';
+    case SVDJob::NONE:
+      return 'N';
+    default:
+      MATX_ASSERT_STR(false, matxInvalidParameter, "Job for SVD not supported");
+      return '\0';
+  }
+}
+
+/* Solver utility functions */
+
+enum class BatchType {
+  Vector = 1,
```

Comment:
Should these be VECTOR and MATRIX to be consistent with the others?

Reply:
Yes, will update